text/0000-autoref-copy.md

```diff
@@ -0,0 +1,200 @@
+methods take references as arguments:
+
+```rust
+use std::collections::HashMap;
+fn main() {
+    let mut map = HashMap::new();
+    map.insert(1, "hello!");
+
+    // We can write this:
+    println!("{:?}", map.get(1));
+
+    // Instead of having to write this:
+    println!("{:?}", map.get(&1));
+}
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+`Copy` types will autoreference: at
+[coercion sites](https://github.com/rust-lang/rfcs/blob/master/text/0401-coercions.md#coercions),
+when a reference is expected, but an owned
+`Copy` type is found, an `&` will be automatically inserted.
+
+If the `Copy` value is a temporary, its lifetime will be promoted to meet the
+required lifetime of the reference (where possible).
+This behavior is specified in detail in
+[RFC 66](https://github.com/rust-lang/rust/issues/15023) and @nikomatsakis's
+[amendment](https://github.com/nikomatsakis/rfcs/blob/rfc66-amendment/text/0066-better-temporary-lifetimes.md).
+
+When it is not possible to promote the `Copy` value's lifetime to the lifetime
+required by the reference, a customized error will be issued:
+
+```rust
+struct u8Ref<'a>(&'a u8);
+
+fn new_static(x: u8) -> u8Ref<'static> {
+    u8Ref(x) // ERROR: borrowed value does not live long enough
+    //    ^autoreference occurs here, but `x` does not live long enough
+}
+```
+
+If the lifetime of the reference would overlap with a mutable reference or a
+mutation of the referenced value, a custom error will be issued:
+
+```rust
+struct u8Ref<'a>(&'a u8);
+
+fn main() {
+    let mut x = 5;
+    let y = u8Ref(x);
+    //            ^ autoreference of `x` occurs here
+    x = 7; // ERROR: cannot assign to `x` because it is borrowed
+}
+```
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- This increases the special behavior of `Copy` types, making it potentially
+confusing to new users. Some simple, non-generic code becomes more confusing
+because `fn foo(x: &i32) { ... }` can now be called like `foo(5)`;
+- The existing coercion mechanism doesn't allow for coercion of generic
+arguments, nor does it use coercions in trait lookup. Because of this, users
+will still need to manually add `&` when trying to, for example, index into
+a `HashMap<usize, T>`
+(users will still need to write `x[&5]` instead of `x[5]`).
+- Autoreferencing may produce surprising errors when attempting to mutate data.
+
+# Rationale and Alternatives
+[alternatives]: #alternatives
+
+One alternative would be to do nothing. However, this issue is frequently
+annoying for new users, and the solution is relatively simple.
+
+Another alternative would be to also add the following conversions for
+`T: Copy`:
+- `T` to `&mut T`: This conversion has the potential to introduce hidden
+mutations. With this change, passing a variable to a function could allow
+the function to change the value of the variable, even if no `&mut` is present.
```

Comment:
I find @RalfJung's logic compelling -- we should consider interior mutability. That said, when we last discussed the Copy question, we had thought to add a lint to reflect the "can copy" vs "should copy" distinction (similar to "must use"). I thought this was a good plan, but it never happened. If it had -- and of course it still can! -- it could certainly apply here.